src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -304,7 +306,9 @@ def cleanRSE(self, rse):
                     try:
                         # execute the actual deletion in bulk - full list of files per directory
                         delResult = []
-                        delResult = ctx.unlink(pfnList)
+
+                        for pfnSlice in list(grouper(pfnList, self.msConfig["filesToDeleteSliceSize"])):
+                            delResult.append(ctx.unlink(pfnSlice))
```

Comment:
Todor, please educate me on the behaviour of this ctx.unlink() call. I understand it will return the value `None` if the file was successfully deleted, right? 
How about the response in case of a failure?
And does the response list maintain the same order of input files provided in the list?

I'm asking these questions to see if there is a better way to account for successful deletions (from L315).

Reply:
Hi @amaltaro , It does return  `None` for any successful deletion, but I do not remember what was the value if otherwise. And I for sure cannot guarantee the order of deletions.  So one thing is for sure, from the last time I was checking  - L315 is the one and only condition to assure that all of the deletions from the bulk list of pfns have been executed successfully.